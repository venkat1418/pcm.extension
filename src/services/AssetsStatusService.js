/* eslint-disable import/prefer-default-export */
import axios from 'axios';

export class AssetsStatusService {

    static check(cmsAuthToken, assetId, userData, pageId){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;
        const lockData = {
            lock: {
                status: null,
                lockedBy: {
                    name: userData.name,
                    email: userData.email
                },
            }
        }
        return axios({
            method: 'post',
            url: '/userlock/check',
            data: {
              'cmsAuthToken':cmsAuthToken,
              'assetId':assetId, 'userId':userData.uid,
              'pageId':pageId, 'lockData':lockData
            },
            baseURL: REACT_APP_PCM_API_BASE_URL
        })
    }

    static unlockThenLock(cmsAuthToken, assetId, userData, pageId){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;
        const lockData = {
            lock: {
                status: null,
                lockedBy: {
                    name: userData.name,
                    email: userData.email
                },
            }
        }

        return axios({
            method: 'post',
            url: '/userlock/lock',
            data: {
              'cmsAuthToken':cmsAuthToken, 'assetId':assetId,
              'userId':userData.uid, 'pageId':pageId, 'lockData':lockData
            },
            baseURL: REACT_APP_PCM_API_BASE_URL
        })
    }

    static removeLock(cmsAuthToken, assetId, userId){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;
        return axios({
            method: 'post',
            url: '/userlock/unlock',
            data: {'cmsAuthToken': cmsAuthToken, 'assetId':assetId, 'userId':userId},
            baseURL: REACT_APP_PCM_API_BASE_URL
        })
    }

}
