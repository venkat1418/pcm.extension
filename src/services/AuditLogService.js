/* eslint-disable import/prefer-default-export */
import axios from 'axios';

export class AuditLogService {

    static logAuditLog(data){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;
        return axios({
            method: 'post',
            url: '/auditlog',
            data,
            baseURL: REACT_APP_PCM_API_BASE_URL
        })
    }
}
