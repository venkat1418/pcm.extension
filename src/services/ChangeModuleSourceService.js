/* eslint-disable import/prefer-default-export */
import axios from 'axios';

export class ChangeModuleSourceService {

    static searchSource(sId){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;
        return axios({
            method: 'get',
            url: `/search/section?nodeId=100746233&query=${sId}`,
            baseURL: REACT_APP_PCM_API_BASE_URL
        })
    }

    static getSourceData(nid){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;

      return  axios({
            method: 'get',
            url: `/capi?nid=${nid}`,
            baseURL: REACT_APP_PCM_API_BASE_URL
        });
    }
}
