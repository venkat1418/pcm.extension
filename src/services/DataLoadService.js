/* eslint-disable import/prefer-default-export */
import axios from 'axios';

export class DataLoadService {
    static getData(nid, template){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;
        return axios({
            method: 'get',
            url: `/config/leftmenu?nid=${nid}&template=${template}`,
            baseURL: REACT_APP_PCM_API_BASE_URL
        })
    }
}
