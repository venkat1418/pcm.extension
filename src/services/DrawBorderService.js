/* eslint-disable import/prefer-default-export */
export class DrawBorderService {

    static Draw(storyid, moduleName){
        const selectedDOM = document.getElementById(`r_${storyid}`);
        if(selectedDOM){
            selectedDOM.classList.add("main_story_boxing");
            const selectedModule = document.getElementById(moduleName);
            if (selectedModule) selectedModule.scrollIntoView({behavior: "instant", block: "center", inline: "nearest"});
        }
    }
    static Erase(storyid){
        const selectedDOM = document.getElementById(`r_${storyid}`);
        if(selectedDOM){
            selectedDOM.classList.remove("main_story_boxing");
        }
    }
}
