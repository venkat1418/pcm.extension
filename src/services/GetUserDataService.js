/* eslint-disable import/prefer-default-export */
import axios from 'axios';

export class GetUserDataService {

    static getUser(cookie){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;
        return axios({
            method: 'post',
            url: '/user',
            data: {'sid': cookie},
            baseURL: REACT_APP_PCM_API_BASE_URL
        });
    }
}
