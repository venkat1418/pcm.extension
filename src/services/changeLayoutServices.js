/* eslint-disable import/prefer-default-export */
import axios from 'axios';

export class changeLayoutServices {

    static getLayoutData(zone){
      const { REACT_APP_PCM_API_BASE_URL } = process.env;
        return axios({
            method: 'get',
            url: `/moduleinfo?zone=${zone}`,
            baseURL: REACT_APP_PCM_API_BASE_URL
        })
    }

}
