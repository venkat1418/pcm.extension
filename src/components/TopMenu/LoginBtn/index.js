// TODO: Move Logout logic to PCM server

import React, { Component } from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import { withCookies } from 'react-cookie';
import { NotificationManager } from 'react-notifications';
import PropTypes from 'prop-types';
import {toggleLogin, toggleEditMode, updateUserData, updateAuthToken} from '../../../actions';
import SetCookieModal from '../../modals/SetCookieModal';
import {GetUserDataService} from '../../../services/GetUserDataService';
import {AuditLogService} from '../../../services/AuditLogService';
import UserAvatar from '../UserAvatar';
import errHandler from '../../../helper/errHandler';


export class LoginBtn extends Component {
    constructor(props){
        super(props);
        this.onLoginWithCookie = this.onLoginWithCookie.bind(this);
        this.state={
            showLoginModal: false
        }
    }

    openLoginModal = ()=>{
        this.setState({ showLoginModal: true });
    }

    closeLoginModal = ()=>{
        this.setState({ showLoginModal: false });
    }

    componentDidMount(){
        if(this.props.cookies.get('toolbeltCookie')){
            const cookie = this.props.cookies.get('toolbeltCookie');
            GetUserDataService.getUser(cookie).then(resp=>{
                if(resp.data.error){
                    NotificationManager.error(`Login failed!`)
                    return false;
                }
                else if(resp.data.uid > 0){
                    const userData = resp.data;
                    this.props.updateUserData(userData);
                    this.props.toggleLogin(true);
                    this.props.updateAuthToken({ newToken: cookie })
                }
            })
            .catch(err=>{ errHandler.notification(err); })
        }
    }

    onLoginWithCookie(cookie, expires){
        if(cookie.trim() === '' || expires.trim() === '') return false;
        GetUserDataService.getUser(cookie).then(resp=>{
            if(resp.data.error){
                NotificationManager.error(`Login failed!`)
                return false;
            }
            else if(resp.data.uid > 0) {
                const userData = resp.data;
                this.props.updateUserData(userData);
                this.props.toggleLogin(true);
                this.props.updateAuthToken({ newToken: cookie })
                const logData = {
                    action: 'user_login',
                    user: userData,
                    updatedAt: new Date().getTime()
                }
                AuditLogService.logAuditLog(logData).then(resp=>{});
                this.onSetCookie(cookie, expires);
                this.closeLoginModal();
            }
        })
        .catch(err=>{ errHandler.notification(err); })
        return false;
    }

    onSetCookie(cookie, expires){
        if(expires){
            this.props.cookies.set('toolbeltCookie', cookie, { path: '/', expires: new Date(expires) });
        }
        else{
            this.props.cookies.set('toolbeltCookie', cookie, { path: '/' });
        }
    }

    render() {
        return (
          <div className="loginComp">
            {
              this.props.isLogin?
                <UserAvatar />
              :
                <div>
                  <button className="loginButton" onClick={this.openLoginModal}>LOGIN</button>
                </div>
          }
            <SetCookieModal
              showModal={this.state.showLoginModal}
              onConfirm={this.onLoginWithCookie}
              onCancel={this.closeLoginModal}
            />
          </div>
        );
    }
};

const mapStateToProps = (state)=>({
        isLogin: state.isLogin,
        isEditMode: state.isEditMode
    })
const mapDispatchToProps = (dispatch)=>bindActionCreators({
        toggleLogin,
        toggleEditMode,
        updateUserData,
        updateAuthToken
    }, dispatch)

LoginBtn.propTypes = {
  cookies: PropTypes.object.isRequired,
  updateUserData: PropTypes.func.isRequired,
  toggleLogin: PropTypes.func.isRequired,
  updateAuthToken: PropTypes.func.isRequired,
  isLogin: PropTypes.bool.isRequired,
}
export default withCookies(connect(mapStateToProps, mapDispatchToProps)(LoginBtn));
