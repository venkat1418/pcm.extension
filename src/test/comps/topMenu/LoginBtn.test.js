
import { shallow } from 'enzyme';
import React from 'react';
import { NotificationManager } from 'react-notifications';
import Cookies from 'universal-cookie'
import LoginBtn from "../../../components/TopMenu/LoginBtn";
import {AuditLogService} from '../../../services/AuditLogService';
import {GetUserDataService} from '../../../services/GetUserDataService';

let cookieHandler = {};

beforeAll(() => {
  cookieHandler = new Cookies({ myCookie: 'pcm' });
  cookieHandler.set('toolbeltCookie', 1234, { path: '/' });
  NotificationManager.error = jest.fn();
});

test('Test the renders of LoginBtn', () => {
   let props = {
      cookies: cookieHandler,
      isLogin: false
   }

   const loginBtnComponent = shallow(<LoginBtn {...props}/>);
   //Create a snapshot with the default render of the LoginBtn component
   expect(loginBtnComponent).toMatchSnapshot();

   //Now check if the name of loginBtn is "LOGIN"
   expect(loginBtnComponent.find('button').at(0).text()).toBe('LOGIN');

  // Simulating the click event on Button
   // at(0) is not actually required as we have only one button being rendered
   loginBtnComponent.find('button').simulate('click');

   /* With the above simulation , we can assert the openCookieModal() callback is triggered
      and state change is occured */
   expect(loginBtnComponent.state().showLoginModal).toBe(true);

   // We can also directly call the asynchrnous methods
   loginBtnComponent.instance().closeLoginModal();
   expect(loginBtnComponent.state().showLoginModal).toBe(false);

   //How ever we will take a snapshot when state has showCookieModal == true
   loginBtnComponent.instance().openLoginModal();
   expect(loginBtnComponent.state().showLoginModal).toBe(true);
   expect(loginBtnComponent).toMatchSnapshot();



   /* This is the final test in scope of this particular component when isLogin is true .
      This is actually passed from store to map the state with props for component , but
      we pass some mocked props to the component directly */
      props = {
         cookies: cookieHandler,
         isLogin: true
      }
    const loginBtnComponent_afterLogin = shallow(<LoginBtn {...props}/>);
    expect(loginBtnComponent_afterLogin).toMatchSnapshot();

});

test.skip('Test the functioning of onLogin callback with Localhost',() => {
  /* InsertCookieModal rendering is not in the scope of this particular as this is shallow
  component test testing . But if we rewiew the snapshot ,we can see the child component
  is accessed with the required props .
  */

  // We should also test the onLogin() which is invoked from the child component .

  // updateUserData and toggleLogin are the actions which
  const props = {
     updateUserData: jest.fn(),
     toggleLogin: jest.fn()
  }

  GetUserDataService.getUser = jest.fn();
  AuditLogService.logAuditLog = jest.fn();

  AuditLogService.logAuditLog
    .mockReturnValue(new Promise(function(resolve, reject){
         resolve({resp: ''});
   }));

   GetUserDataService.getUser
      .mockReturnValueOnce(new Promise(function(resolve, reject){
         resolve({'data' : {'uid' : '99'}});
      }))
      .mockReturnValueOnce(new Promise(function(resolve, reject){
         resolve({'data' : {'error' : '403 error'}});
      }))

  const loginBtnComponent = shallow(<LoginBtn {...props}/>);
  // manually setting the showCookieModal to true to observe the change in behavior futher
   loginBtnComponent.setState({showCookieModal:true});

  // onLogin is called with empty cookie
  loginBtnComponent.instance().onLogin('');

  // when called with empty cookie execution is just returned without invoking any actions
  expect(props.updateUserData.mock.calls.length).toBe(0)
  expect(props.toggleLogin.mock.calls.length).toBe(0)
 // Modal should be still open
  expect(loginBtnComponent.state().showCookieModal).toBe(true);

// Now use the test cookie (1234)
  loginBtnComponent.instance().onLogin("1234");

  // actions should be invoked
  expect(props.updateUserData.mock.calls.length).toBe(1)
  expect(props.toggleLogin.mock.calls.length).toBe(1)

// actions should be invoked with the expected data
  expect(props.updateUserData).toHaveBeenCalledWith({"email": "local.narenda@cnbc.com", "name": "local narenda", "roles": ["admin", "mpsadmin", "administrator"], "uid": 100000898})
  expect(props.toggleLogin).toHaveBeenCalledWith(true);
// Modal should close
  expect(loginBtnComponent.state().showCookieModal).toBe(false);


});

test('Test the functioning of onLoginWithCookie callback with mocked user', done => {
  const props = {
     updateUserData: jest.fn(),
     toggleLogin: jest.fn(),
     updateAuthToken: jest.fn(),
     cookies: cookieHandler,
  }

  GetUserDataService.getUser = jest.fn();
  AuditLogService.logAuditLog = jest.fn();

  AuditLogService.logAuditLog
    .mockReturnValue(new Promise(function(resolve, reject){
         resolve({resp: ''});
   }));

   GetUserDataService.getUser
      .mockReturnValue(new Promise(function(resolve, reject){
         resolve({'data' : {'uid' : '99'}});
      }))

  const loginBtnComponent = shallow(<LoginBtn {...props}/>);
  // manually setting the showCookieModal to true to observe the change in behavior futher
   loginBtnComponent.setState({showCookieModal:true});

// this shoudl return false
  loginBtnComponent.instance().onLoginWithCookie('','');
  // this should not return false

  loginBtnComponent.instance().onLoginWithCookie(cookieHandler.get('toolbeltCookie'),'false');

/* Implementing the asynchronous fashion onLogin() has promises execution test cases
should not execute as soon as onLogin return .
*/
  setTimeout(() => {
      expect(props.toggleLogin.mock.calls.length).toBe(2)
      expect(props.updateUserData.mock.calls.length).toBe(2)
      expect(props.updateAuthToken.mock.calls.length).toBe(2)
      done();
    }, 10);
});


test('Test the functioning of onLogin callback with login error', done => {
  const props = {
     updateUserData: jest.fn(),
     toggleLogin: jest.fn(),
     cookies: cookieHandler,
  }

  window.alert = jest.fn();

  GetUserDataService.getUser = jest.fn();
  AuditLogService.logAuditLog = jest.fn();

  AuditLogService.logAuditLog
    .mockReturnValue(new Promise(function(resolve, reject){
         resolve({resp: ''});
   }));

   GetUserDataService.getUser
      .mockReturnValue(new Promise(function(resolve, reject){
         resolve({'data' : {'error' : '403 error'}});
      }))

  const loginBtnComponent = shallow(<LoginBtn {...props}/>);
  // manually setting the showCookieModal to true to observe the change in behavior futher
   loginBtnComponent.setState({showCookieModal:true});

// Further to mock the GetUserDataService and write tests where a mocked non localtest should be retrieved
  loginBtnComponent.instance().onLoginWithCookie(cookieHandler.get('toolbeltCookie'),'false');

  setTimeout(() => {
      expect(props.toggleLogin.mock.calls.length).toBe(0)
      expect(props.updateUserData.mock.calls.length).toBe(0)
    //  expect(window.alert).toHaveBeenCalledWith("login failed!");
      expect(NotificationManager.error).toHaveBeenCalledWith("Login failed!")
      done();
    }, 10);

});
