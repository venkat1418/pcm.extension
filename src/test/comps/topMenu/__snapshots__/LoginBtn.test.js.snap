// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the renders of LoginBtn 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginBtn
    cookies={
      Cookies {
        "HAS_DOCUMENT_COOKIE": true,
        "cookies": Object {
          "toolbeltCookie": "1234",
        },
        "hooks": undefined,
      }
    }
    isLogin={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <button
            className="loginButton"
            onClick={[Function]}
          >
            LOGIN
          </button>
        </div>,
        <SetCookieModal
          onCancel={[Function]}
          onConfirm={[Function]}
          showModal={false}
        />,
      ],
      "className": "loginComp",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="loginButton"
            onClick={[Function]}
          >
            LOGIN
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "LOGIN",
            "className": "loginButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onCancel": [Function],
          "onConfirm": [Function],
          "showModal": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <button
              className="loginButton"
              onClick={[Function]}
            >
              LOGIN
            </button>
          </div>,
          <SetCookieModal
            onCancel={[Function]}
            onConfirm={[Function]}
            showModal={false}
          />,
        ],
        "className": "loginComp",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="loginButton"
              onClick={[Function]}
            >
              LOGIN
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "LOGIN",
              "className": "loginButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "LOGIN",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onCancel": [Function],
            "onConfirm": [Function],
            "showModal": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test the renders of LoginBtn 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginBtn
    cookies={
      Cookies {
        "HAS_DOCUMENT_COOKIE": true,
        "cookies": Object {
          "toolbeltCookie": "1234",
        },
        "hooks": undefined,
      }
    }
    isLogin={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <button
            className="loginButton"
            onClick={[Function]}
          >
            LOGIN
          </button>
        </div>,
        <SetCookieModal
          onCancel={[Function]}
          onConfirm={[Function]}
          showModal={true}
        />,
      ],
      "className": "loginComp",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="loginButton"
            onClick={[Function]}
          >
            LOGIN
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "LOGIN",
            "className": "loginButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onCancel": [Function],
          "onConfirm": [Function],
          "showModal": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <button
              className="loginButton"
              onClick={[Function]}
            >
              LOGIN
            </button>
          </div>,
          <SetCookieModal
            onCancel={[Function]}
            onConfirm={[Function]}
            showModal={true}
          />,
        ],
        "className": "loginComp",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="loginButton"
              onClick={[Function]}
            >
              LOGIN
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "LOGIN",
              "className": "loginButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "LOGIN",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onCancel": [Function],
            "onConfirm": [Function],
            "showModal": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test the renders of LoginBtn 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginBtn
    cookies={
      Cookies {
        "HAS_DOCUMENT_COOKIE": true,
        "cookies": Object {
          "toolbeltCookie": "1234",
        },
        "hooks": undefined,
      }
    }
    isLogin={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withCookies(Connect(UserInfo)) />,
        <SetCookieModal
          onCancel={[Function]}
          onConfirm={[Function]}
          showModal={false}
        />,
      ],
      "className": "loginComp",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onCancel": [Function],
          "onConfirm": [Function],
          "showModal": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withCookies(Connect(UserInfo)) />,
          <SetCookieModal
            onCancel={[Function]}
            onConfirm={[Function]}
            showModal={false}
          />,
        ],
        "className": "loginComp",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onCancel": [Function],
            "onConfirm": [Function],
            "showModal": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
